src/shared/codelens/localLambdaRunner.ts

```diff
@@ -244,14 +244,19 @@ export class LocalLambdaRunner {
             invoker: this.localInvokeCommand
         })
 
-        const startInvokeTime = new Date()
-        await command.execute()
+        const timelimit = this.configuration.readSetting<number>(
+            'samcli.debug.attach.timeout.millis',
+            SAM_LOCAL_PORT_CHECK_RETRY_TIMEOUT_MILLIS_DEFAULT
+        )
+        const timer = new InvokeTimer(timelimit ? timelimit : SAM_LOCAL_PORT_CHECK_RETRY_TIMEOUT_MILLIS_DEFAULT)
```

Comment:
We shouldn't need to check whether `timelimit` is truthy, because we provided a default to `readSetting`. You can allow the compiler to reason about this by updating the signature of `readSetting` from

```typescript
export interface SettingsConfiguration {
    readSetting<T>(settingKey: string, defaultValue?: T): T | undefined

    ...
}

export class DefaultSettingsConfiguration implements SettingsConfiguration {
    ...

    public readSetting<T>(settingKey: string, defaultValue?: T): T | undefined {
        ...
    }

    ...
}
```

to

```typescript
export interface SettingsConfiguration {
    readSetting<T>(settingKey: string): T | undefined
    readSetting<T>(settingKey: string, defaultValue: T): T

    ...
}

export class DefaultSettingsConfiguration implements SettingsConfiguration {
    ...

    public readSetting<T>(settingKey: string): T | undefined
    public readSetting<T>(settingKey: string, defaultValue: T): T
    public readSetting<T>(settingKey: string, defaultValue?: T): T | undefined {
        // Unchanged implementation.
    }

    ...
}
```